File: src/main/java/org/antop/java/chapter3/StringBuilderExtension.java
package org.antop.java.chapter3;

public class StringBuilderExtension {
    public static char getLastChar(StringBuilder sb) {
        return sb.charAt(sb.length() - 1);
    }

    public static void setLastChar(StringBuilder sb, char ch) {
        sb.setCharAt(sb.length() - 1, ch);
    }

    public static void main(String[] args) {
        var sb = new StringBuilder("Antop!");
        System.out.println(sb); // Antop!
        setLastChar(sb, '?');
        System.out.println(sb); // Antop?

    }

}

File: src/main/kotlin/org/antop/kotlin/chapter3/StringBuilderExtension.kt
package org.antop.kotlin.chapter3

var StringBuilder.lastChar: Char
    get() = this[length - 1]
    set(ch) {
        this.setCharAt(length - 1, ch)
    }

fun main() {
    val sb = StringBuilder("Antop!")
    println(sb) // Antop!
    sb.lastChar = '?'
    println(sb) // Antop?
}
